src/control/common/meminfo.go

```diff
@@ -59,12 +94,43 @@ func parseInt(a string, i *int) {
 	*i = v
 }
 
-func parseMemInfo(input io.Reader) (*MemInfo, error) {
-	mi := new(MemInfo)
+func processNodeLine(nodeStr, txt string, mi *MemInfoT) (string, error) {
+	idStr := strings.Split(nodeStr, " ")[1]
+	id, err := strconv.Atoi(idStr)
+	if err != nil {
+		return "", err
+	}
+
+	if mi.NumaNodeID == -1 {
+		mi.NumaNodeID = id
+	} else if mi.NumaNodeID != id {
+		return "", errors.New("unexpected mix of node ids in meminfo file")
+	}
+
+	return strings.Replace(txt, nodeStr, "", 1), nil
+}
+
+func parseMemInfoT(input io.Reader) (*MemInfoT, error) {
+	mi := new(MemInfoT)
+
+	mi.NumaNodeID = -1
+	re, err := regexp.Compile(`Node \d+ `)
```

Comment:
minor nit: Doesn't matter in this case as the code isn't called often, but you could make the compiled regexp a package variable so that it's compiled once.

Also, since you're hard-coding the pattern, there's no point in checking for an error. You could use `regexp.MustCompile()` and save a few lines of code.

Reply:
will update as I have to fix copyrights, done